### PR TITLE
feat: replace eudev with systemd-udevd

### PR DESCRIPTION
**Pull request by dsseng**
# Pull Request

## What? (description)
replace eudev with systemd-udevd

## Why? (reasoning)
eudev has been abandoned by Gentoo team, Gentoo itself transitioning to mdev (incompatible with udev rules) or systemd-udevd, which can now be built and used independently of systemd PID1 service manager and other parts of the systemd stack. Chances are it will also help us with SELinux support (running udev rules for relabeling device nodes)

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [ ] you ran conformance (`make conformance`)
- [ ] you formatted your code (`make fmt`)
- [ ] you linted your code (`make lint`)
- [ ] you generated documentation (`make docs`)
- [ ] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

### CODE REVIEW

**Comment by dsseng** [comment]
Will need to be tested on hardware and with some hardware using udev rules to be sure everything runs well

**Comment by dsseng** [comment]
CI fails due to lack of arm64 build. Will be available once pkgs PR gets merged

File: internal/app/machined/pkg/system/services/udevd.go [hunk 1]
```diff
@@ -72,12 +72,13 @@ func (c *Udevd) Runner(r runtime.Runtime) (runner.Runner, error) {
...
 		ProcessArgs: []string{
-			"/sbin/udevd",
+			"/sbin/systemd-udevd",
 			"--resolve-names=never",
+			"-D",
 		},
 	}
 
-	debug := false
+	debug := true
```
Comment (smira) [inline internal/app/machined/pkg/system/services/udevd.go:81]: let's not forget to revert this, and PKG_ variables

**Comment by smira** [comment]
and your packages are private, please make them public to test it out

**Comment by dsseng** [comment]
Oops, libcap was private. Fixed

**Review by smira — APPROVED** [review]
:+1:  as long as it passes CI

**Pull request merged**